src/contentscanners/icapscan.cpp

```diff
@@ -398,20 +305,12 @@ int icapinstance::scanFile(HTTPHeader *requestheader, HTTPHeader *docheader, con
     char *object = new char[100];
     int objectsize = 0;
 
-#ifndef NEWDEBUG_OFF
-        if(o.myDebug->ICAPC)
-        {
-            std::ostringstream oss (std::ostringstream::out);
-            oss << thread_id << "About to send file data to icap" << std::endl;
-            o.myDebug->Debug("ICAPC",oss.str());
-            std::cerr << thread_id << "About to send file data to icap" << std::endl;
-        }
-#endif
+    DEBUG_avscan("About to send file data to icap");
 
     if (usepreviews && (filesize > previewsize)) {
         try {
             while (sent < previewsize) {
-                int rc = readEINTR(filefd, data, previewsize);
+                int rc = read(filefd, data, previewsize);
```

Comment:
Codacy found an issue: [Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).  ](https://app.codacy.com/manual/numsys/e2guardian/pullRequest?prid=6238852)